### DISCUSSION TITLE
How to stop a cluster correctly?

### DISCUSSION THREAD

**Discussion by ldapomni** [seaweedfs/seaweedfs#2122] [Q&A]

**Comment by chrislusf** [comment]
Currently, ctrl+c is enough.

**Reply by ldapomni** [reply]
May be it will add weed shell [shutdown], for the correct done all processes?

**Reply by NyaMisty** [reply]
It's not quite convincing :(
What if there's still master maintenance script running? For example, there's a volume.balance operation running, and what if I ctrl-c during that? Will it ABSOULUTELY functions correctly? 
Also what's the order of shutting down?  Ctrl-C everything at the same time? Won't that causing racing condition?
Or filer -> volume -> master?

**Reply by SmsS4** [reply]
We are using seaweed for almost 1 year and we shutdown master many times during `fix.replication`, and `vacuum` and never had any problem with it.
